src/NFTDriver.sol

```diff
@@ -55,7 +63,30 @@ contract NFTDriver is ERC721Burnable, ERC2771Context, Managed {
     /// @notice Get the ID of the next minted token.
     /// @return tokenId The token ID.
     function nextTokenId() public view returns (uint256 tokenId) {
-        return (uint256(driverId) << 224) + StorageSlot.getUint256Slot(_mintedTokensSlot).value;
+        return calcTokenIdWithSalt(address(0), _nftDriverStorage().mintedTokens);
+    }
+
+    /// @notice Calculate the ID of the token minted with salt.
+    /// @param minter The minter of the token.
+    /// @param salt The salt used for minting the token.
+    /// @return tokenId The token ID.
+    function calcTokenIdWithSalt(address minter, uint64 salt)
+        public
+        view
+        returns (uint256 tokenId)
+    {
+        tokenId = driverId;
+        tokenId = (tokenId << 160) | uint160(minter);
```

Comment:
I think it would be nice to use constants for the offsets

Reply:
I've tried different approaches and this one seemed the cleanest and most concise. Could you propose changes with the alternative design?

Reply:
```solidity
uint internal constant ADDRESS_OFFSET = 160;
```
```solidity
 tokenId = (tokenId << ADDRESS_OFFSET) | uint160(minter);
```

Reply:
Is it cleaner or more bug-resistant? To me it seems noisy and in non-local, we get an extra constant to look at and you need to find the value of `ADDRESS_OFFSET` to see if the code makes sense. The expression `(tokenId << 160) | uint160(minter);` is self-contained and obviously correct, shift by 160 bits and set 160 bits. If we needed to repeat this logic in multiple places, I'd agree, a constant would be a good idea, but here it's just this one place.

Reply:
In my opinion, it makes the code easier to understand.  Currently, you leave it as an exercise to the reader.
The reader needs to realize, ah 160 bits are the offset for an address.

It is easier in my opinion if I can read the interpretation of a number.  

This is my reasoning behind it.

Reply:
If it's unclear, maybe we should comment the shifting operations? The solution with the constant only adds one piece of information, that 160 is the "address offset", but it spreads it across the entire contract, a comment would be available locally.

Reply:
I think we should add a comment. We had C4 auditors who struggled with bit manipulation and shifting. A lot of frontend dev's might be also not so familiar with these low-level concepts.

Reply:
Ok, I'll add docs in the next PR, there are a few places with bit shifting, so it makes sense to comment them all.